src/transaction.js

```diff
@@ -51,13 +68,24 @@ Transaction.fromBuffer = function (buffer, __noStrict) {
   var tx = new Transaction()
   tx.version = readUInt32()
 
+  var marker = buffer.readUInt8(offset)
+  var flag = buffer.readUInt8(offset + 1)
+
+  var hasWitnesses = false
+  if (marker === Transaction.ADVANCED_TRANSACTION_MARKER &&
+      flag === Transaction.ADVANCED_TRANSACTION_FLAG) {
+    offset += 2
+    hasWitnesses = true
+  }
+
   var vinLen = readVarInt()
   for (var i = 0; i < vinLen; ++i) {
     tx.ins.push({
       hash: readSlice(32),
       index: readUInt32(),
       script: readScript(),
-      sequence: readUInt32()
+      sequence: readUInt32(),
+      witness: EMPTY_SCRIPT
```

Comment:
Oh - also, that was probably intentional, because segwit is a pathway to introducing script types other than bitcoin script. So those Buffers we extract from witness data could wind up in a crazy new segwit version.